src/GraphQL/Utilities/Visitors/SchemaValidationVisitor.cs

```diff
@@ -152,33 +155,58 @@ public override void VisitInputObject(IInputObjectGraphType type, ISchema schema
         }
 
         /// <inheritdoc/>
-        public override void VisitInputFieldDefinition(FieldType field, IInputObjectGraphType type, ISchema schema)
+        public override void VisitInputObjectFieldDefinition(FieldType field, IInputObjectGraphType type, ISchema schema)
         {
             // 2.2
             if (field.Name.StartsWith("__"))
                 throw new InvalidOperationException($"The input field '{field.Name}' of an Input Object '{type.Name}' must not have a name which begins with the __ (two underscores).");
 
+            if (field.ResolvedType == null)
+                throw new InvalidOperationException($"The field '{field.Name}' of an Input Object type '{type.Name}' must have non-null '{nameof(IFieldType.ResolvedType)}' property.");
+
             // 2.3
-            if (field.ResolvedType != null ? field.ResolvedType.IsInputType() == false : field.Type?.IsInputType() == false)
```

Comment:
@Shane32 I copied this line from `ComplextGraphType.AddField`. The second part of the check is unnecessary, because by this point `ResolvedType` should always be set.
![изображение](https://user-images.githubusercontent.com/21261007/109512450-2a2a0c00-7ab5-11eb-8c0b-f46072a0322b.png)